### PR TITLE
Update kubevirtCloudSpec to inject infraStorageClass info

### LINKED ISSUE #8546: Initialize a standard storage class for user cluster
We have to support storage class initialization on the user cluster that customers can use hot-pluggable disks.

### PR DESCRIPTION
**Pull request by sankalp-r**
**What does this PR do / Why do we need it**:
This PR introduces a new field `InfraStorageClasses` in `KubevirtCloudSpec` which holds the names of all the Kubevirt Infra-StorageClasses. Later this info is passed to `csi-driver-opeartor.yaml` for `kubevirt-csi-driver-operartor` ADDON creation.

**Does this PR close any issues?**:
Fixes #8546 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Added support for storage classes initialization on KubeVirt user clusters that customers can use hot-pluggable disks.
```

Signed-off-by: Sankalp Rangare <sankalprangare786@gmail.com>

### CODE REVIEW

File: pkg/provider/cloud/kubevirt/storage.go [hunk 1]
```diff
@@ -22,23 +22,30 @@ import (
...
 	utilpointer "k8s.io/utils/pointer"
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var (
-	csiServiceAccountNamespace = metav1.NamespaceDefault
-	csiResourceName            = "kubevirt-csi"
+	csiServiceAccountNamespace  = metav1.NamespaceDefault
+	csiResourceName             = "kubevirt-csi"
+	InfraStorageClassAnnotation = "kubevirt-initialization.k8c.io/initialize-sc"
```
Comment (mfranczy) [inline pkg/provider/cloud/kubevirt/storage.go]: please make this a constant

**Review by mfranczy — CHANGES_REQUESTED** [review]
overall looks good, just a small nit

**Comment by mfranczy** [comment]
@sankalp-r please add unit tests as well, we have to finally start writing tests :)

**Review by mfranczy — APPROVED** [review]
/lgtm

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve ⚡

**Comment by ahmedwaleedmalik** [comment]
/retest
Flake

**Pull request merged**